MatrixKit/Models/Contact/MXKContactManager.m

```diff
@@ -1523,7 +1523,7 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
             [self internationalizePhoneNumbers:[[MXKAppSettings standardAppSettings] phonebookCountryCode]];
             
             // Refresh local contacts if we have some
-            if (!self->localContactByContactID)
+            if (self->localContactByContactID.count)
```

Comment:
Is it worth also checking that the user hasn't turned off contact sync recently too?
```suggestion
            if (MXKAppSettings.standardAppSettings.syncLocalContacts && self->localContactByContactID.count)
```